construct_shuffled_test_splits.py

```diff
@@ -0,0 +1,49 @@
+import random
+from typing import List
+from sumtool.storage import get_summary_metrics
+import json
+from src.data_utils import load_xsum_dict, get_gold_xsum_data
+
+
+def construct_test_split(xsum_test, filter_fn) -> List[str]:
+    """
+    Construct shuffled test set from xent test,
+    filter_fn indicates how to filter xsum test.
+    """
+    filtered_sum_ids = [sum_id for sum_id in xsum_test.keys() if filter_fn(sum_id)]
+    rng_data_split = random.Random(42)
+    return rng_data_split.sample(filtered_sum_ids, len(filtered_sum_ids))
+
+
+if __name__ == "__main__":
+    xsum_test = load_xsum_dict("test")
+    baseline_metadata = get_summary_metrics("xsum", "facebook-bart-large-xsum")
+    gold_sums, gold_metadata = get_gold_xsum_data()
+
+    def filter_xsum(sum_id):
+        return "xent-train" not in gold_metadata[sum_id]
+
+    def filter_xent_test(sum_id):
+        return "xent-test" in gold_metadata[sum_id]
+
+    def filter_xent_extrinsic(sum_id):
+        return (
+            "xent-train" not in gold_metadata[sum_id]
```

Comment:
in addition to ensuring that the example is not in `xent-train`, shouldn't we make sure that the example is within `xent-test?` At least, that's what I would presume from a function name of `filter_xent_extrinsic`

Reply:
or maybe we should rename the function -- if i remember correctly, this is our legacy test set, which might include examples included in xent-test or not (but all are in xsum-test)

Reply:
My bad, this function name is incorrect - it should be `filter_xsum_extrinsic`